soundata/core.py

```diff
@@ -361,6 +377,459 @@ def validate(self, verbose=True):
         )
         return missing_files, invalid_checksums
 
+    @lru_cache(maxsize=None)  # Setting maxsize to None for an unbounded cache
+    def compute_clip_statistics(self):
+        durations = [
+            len(self.clip(c_id).audio[0]) / self.clip(c_id).audio[1]
+            for c_id in tqdm(
+                list(self._index["clips"].keys())[:100], desc="Calculating durations"
+            )
+        ]
+
+        # Calculate statistics
+        total_duration = sum(durations)
+        mean_duration = np.mean(durations)
+        median_duration = np.median(durations)
+        std_deviation = np.std(durations)
+        min_duration = np.min(durations)
+        max_duration = np.max(durations)
+
+        return {
+            "durations": durations,
+            "total_duration": total_duration,
+            "mean_duration": mean_duration,
+            "median_duration": median_duration,
+            "std_deviation": std_deviation,
+            "min_duration": min_duration,
+            "max_duration": max_duration,
+        }
+
+    def plot_clip_durations(self):
+        stats = self.compute_clip_statistics()
+        durations = stats["durations"]
+        total_duration = stats["total_duration"]
+        mean_duration = stats["mean_duration"]
+        median_duration = stats["median_duration"]
+        std_deviation = stats["std_deviation"]
+        min_duration = stats["min_duration"]
+        max_duration = stats["max_duration"]
+
+        # Determine unit conversion (seconds or minutes)
+        convert_to_minutes = mean_duration > 60 or median_duration > 60
+        conversion_factor = 60 if convert_to_minutes else 1
+        unit = "minutes" if convert_to_minutes else "seconds"
+
+        durations = [d / conversion_factor for d in durations]
+        mean_duration /= conversion_factor
+        median_duration /= conversion_factor
+        total_duration /= conversion_factor
+
+        # Convert total duration to hours if it exceeds 120 minutes
+        if convert_to_minutes:
+            if total_duration > 120:
+                total_duration /= 60
+                total_duration_unit = "hours"
+            else:
+                total_duration_unit = "minutes"
+        else:
+            if total_duration > 120:
+                total_duration /= 60
+                total_duration_unit = "minutes"
+            else:
+                total_duration_unit = "seconds"
+
+        # Define the base colors
+        base_colors = ["#404040", "#126782", "#C9C9C9"]
```

Comment:
Do we only need three colors for all datasets?

Reply:
Colors to resemble the soundata template